### ISSUE TITLE
Image download not retried, progressBlock and completionHandler not called after an error.

### ISSUE THREAD

**Issue by alenofx** [onevcat/Kingfisher#49]
When an error happened (like after a cancelled task) the image download at the same URL will not be retried; the progress and completion callbacks are not executed but just appended to "fetchLoads".

![fetchloads](https://cloud.githubusercontent.com/assets/867766/7486548/437651fa-f3a7-11e4-99b4-2be6a638d68d.png)

Probably a "cleanForURL" call is missing from the NSURLSessionTaskDelegate:

```swift
public func URLSession(session: NSURLSession, task: NSURLSessionTask, didCompleteWithError error: NSError?) {
    if let URL = task.originalRequest.URL {
        if let error = error { // Error happened
            callbackWithImage(nil, error: error, imageURL: URL)
            cleanForURL(URL) // <---
        } else { //Download finished without error
```

**Comment by onevcat** [comment]
Definitely! Thanks for pointing it out. I will fix it soon.

**Comment by onevcat** [comment]
Should be fixed in #50

**Comment by alenofx** [comment]
Thank you, it's fixed.
